### DISCUSSION TITLE
Audience and their needs and objective of foundation material: first steps

### DISCUSSION THREAD

**Discussion by kitmacleod** [ProjectPythia/projectpythia.github.io#141] [General]
Great to see your tweet about the 'foundations' material. You will have considered this and will collectively have a clear idea of audience and purpose (and a lot more Python/programming teaching experience than me). If not already considered, then using a sample of intended audience/users then get their feedback on current structure and content (also useful to know their programming/data science backgrounds; which will be varied, even in the US). If truly going from 'zero' then I would suggest very careful consideration of this onboarding and initial steps (e.g. before getting overwhelmed by Matplotlib's API; respect to John Hunter and many others). Yesterday, I came across this useful approach (from the Julia community) [Is this course for me? Julia Programming For Nervous Beginners](https://www.youtube.com/watch?v=ub3tqCWZmo4&t=1040s) "Our philosophy can be summed up in the slogan 'Take small steps, leave no gaps, everything makes sense" (P.S. I have not gone through the series to assess if they deliver this). Related to this key onboarding, is to consider a short (10 min) video as many learners benefit from that format, in part as it can provide key info missing from text. An example from a master (of developer relations) is I noticed MongoDB are increasingly providing blog posts in text and video format (quite a commitment/overhead).

**Comment by clyne** [comment]
+1 for "Take small steps, leave no gaps, everything makes sense" and videos. Stay tuned!
